unittests/src/main/java/uk/ac/manchester/tornado/unittests/batches/TestBatches.java

```diff
@@ -76,8 +77,19 @@ public static void compute(short[] arrayA, short[] arrayB, short[] arrayC) {
     @Test
     public void test100MB() {
 
+        long maxAllocMemory = getTornadoRuntime().getDefaultDevice().getDeviceContext().getMemoryManager().getHeapSize();
+
+        //check if there is enough memory for at least one chunk
+        if (maxAllocMemory < 100 * 1024 * 1024) {
+            throw new UnsupportedConfigurationException("Not enough memory to run the test");
+        }
+
         // Fill 800MB of float array
         int size = 200000000;
+        // or as much as we can
+        if (size * 4 > maxAllocMemory) {
+            size = (int) ((maxAllocMemory / 4 / 2) * 0.9);
```

Comment:
I miss a bit of context here, what are the numbers we divide maxAllocMemory?

Reply:
I had to do like this, since in some tests we fill one array, in some tests two, in some three.. And we have heap. I always multiply by 0.9 not to fill all the memory.